packages/miniflare/src/plugins/kv/assets.ts

```diff
@@ -92,8 +99,184 @@ export function getAssetsServices(options: AssetsOptions): Service[] {
 					name: SharedBindings.MAYBE_SERVICE_BLOBS,
 					service: { name: storageServiceName },
 				},
+				{
+					name: "__STATIC_ASSETS_REVERSE_MAP",
+					data: assetsReverseMap,
+				},
 			],
 		},
 	};
 	return [storageService, namespaceService];
 }
+
+/**
+ *  ASSET MANIFEST
+ *  1. Traverse the asset directory to create an asset manifest.
+ * 		(In prod the manifest contains a pathHash and a contentHash. The
+ * 		contentHash is used for uploading and as the keys for the KV namespace
+ * 		where the assets are stored. Uploading is irrelevant in dev, so for
+ * 		performance reasons, the pathHash is reused for the "contentHash".)
+ *  2. Sort and binary encode the asset manifest
+ * 	This is available to asset service worker as a binding.
+ */
+const MAX_ASSET_COUNT = 20_000;
+const MAX_ASSET_SIZE = 25 * 1024 * 1024;
+const MANIFEST_HEADER_SIZE = 20;
+
+const PATH_HASH_OFFSET = 0;
+const PATH_HASH_SIZE = 16;
+
+const CONTENT_HASH_OFFSET = PATH_HASH_SIZE;
+const CONTENT_HASH_SIZE = 16;
+
+const TAIL_RESERVED_SIZE = 8;
+
+const ENTRY_SIZE = PATH_HASH_SIZE + CONTENT_HASH_SIZE + TAIL_RESERVED_SIZE;
+
+const walk = async (dir: string) => {
+	const files = await fs.readdir(dir, { recursive: true });
+	const manifest: Uint8Array[] = [];
+	let counter = 0;
+	await Promise.all(
+		files.map(async (file) => {
+			const filepath = path.join(dir, file);
+			const relativeFilepath = path.relative(dir, filepath);
+			const filestat = await fs.stat(filepath);
+
+			if (filestat.isSymbolicLink() || filestat.isDirectory()) {
+				return;
+			} else {
+				if (counter >= MAX_ASSET_COUNT) {
+					throw new Error(
+						`Maximum number of assets exceeded.\n` +
+							`Cloudflare Workers supports up to ${MAX_ASSET_COUNT.toLocaleString()} assets in a version. We found ${counter.toLocaleString()} files in the specified assets directory "${dir}".\n` +
+							`Ensure your assets directory contains a maximum of ${MAX_ASSET_COUNT.toLocaleString()} files, and that you have specified your assets directory correctly.`
+					);
+				}
+
+				if (filestat.size > MAX_ASSET_SIZE) {
+					throw new Error(
+						`Asset too large.\n` +
+							`Cloudflare Workers supports assets with sizes of up to ${prettyBytes(
+								MAX_ASSET_SIZE,
+								{
+									binary: true,
+								}
+							)}. We found a file ${filepath} with a size of ${prettyBytes(
+								filestat.size,
+								{
+									binary: true,
+								}
+							)}.\n` +
+							`Ensure all assets in your assets directory "${dir}" conform with the Workers maximum size requirement.`
+					);
+				}
+
+				manifest.push(await hashPath(encodeFilePath(relativeFilepath)));
```

Comment:
A manifest entry consist of a path hash (16 bytes) and a content hash (16 bytes), right?
In the case of Miniflare they are going to be the same thing, since we are not bothering to hash the contents of the file.
So I would expect this to be:

```suggestion
        const pathHash = await hashPath(encodeFilePath(relativeFilepath));
				manifest.push(pathHash, pathHash);
```

Or am I missing something here?

Reply:
Think I'm doing essentially the same thing when I insert in the pathHash twice while encoding here: https://github.com/cloudflare/workers-sdk/blob/1b9f58546fb7165882c7c6ba551f1e0a8d739397/packages/miniflare/src/plugins/kv/assets.ts#L230